worker/src/socket.rs

```diff
@@ -79,6 +75,7 @@ impl Socket {
             read: None,
             write: None,
             close: None,
+            opened: None,
```

Comment:
I'm not sure that this is necessary, the read / write / close properties I think are used for locking / unlocking the internal reader of the socket. Whereas this never gets set / accessed I think?

Reply:
What tests would you like me to write? A general list is okay.

Reply:
Unfortunately I'm not sure it will be straightforward to mock sockets. We didn't do any tests when we first added this API. 

But testing for "not opened" might be the easiest. Just open a socket to somewhere that shouldn't be valid (maybe some random localhost port) and verify that opened promise resolves and indicates failure?

Reply:
I attempted a test like so:

```
#[test]
    fn test_is_opened() {
        println!("hi im running the special test :)");
        let socket = Socket::builder()
        .secure_transport(SecureTransport::StartTls)
        .connect("database_url", 5432);

        assert!(true);
    }
```

and saw that this test has the same error as one of the two failing tests, `cannot call wasm-bindgen imported functions on non-wasm targets`. Zeb mentioned you guys are going to work on adding Vitest for Workers to workers-rs, so I'm assuming getting these working isn't a high priority?

I made an attempt and I am honestly not sure about being able to get these working without a lot of hassle.

Reply:
We already use vitest / miniflare. Check out the `worker-sandbox` directory. 

If you run `npm install` and `npm run test` then it should run the test suite.

Reply:
Gotcha. The Miniflare API doesn't provide access to TCP sockets as far as I know. I think you were on the money with sockets being extremely difficult to test, I don't see a way to accomplish this without a lot of API changes. I'd be happy to take your suggestion if you have any ideas.

Reply:
I'll propose something in a follow up PR